docker/ci/github/Dockerfile

```diff
@@ -5,24 +5,23 @@ COPY rust-toolchain /diem/rust-toolchain
 COPY cargo-toolchain /diem/cargo-toolchain
 COPY scripts/dev_setup.sh /diem/scripts/dev_setup.sh
 
+#this is the default home on docker images in gha, until it's not?
+ENV HOME "/github/home"
+#Needed for sccache to function
+ENV CARGO_HOME "/opt/cargo/"
+
 # Batch mode and all operations tooling
-RUN /diem/scripts/dev_setup.sh -t -o -b -p -y && apt-get clean && rm -rf /var/lib/apt/lists/*
-ENV PATH "/root/.cargo/bin:/root/bin/:$PATH"
-
-# Since proper use of sccache demands the exact same path for source and crates, configure it so.
-# Paths that every use can choose to setup.
-RUN mkdir -p /opt/cargo/
-RUN mkdir -p /opt/git/
-
-# Since github actions may not run as root in the docker image, and it's behavior
-# is undocument, and all request for information about any invarients are not responded to....
-# So we may symlink this locations to whatever home directory gha creates, currently /github/home/
-RUN chmod -R o+rwx /root/bin
-RUN chmod -R o+rwx /root/.dotnet
-RUN chmod -R o+rwx /root/.rustup
-RUN chmod -R o+rwx /root/.cargo
-RUN chmod -R o+rwx /root/.local
-RUN chmod -R o+rwx /root/.nuget
+RUN mkdir -p /github/home \
+    && mkdir -p /opt/cargo/ \
+    && mkdir -p /opt/git/ \
+    && /diem/scripts/dev_setup.sh -t -o -b -p -y \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
```

Comment:
why the refactor into a single batched command?

Reply:
to get a single docker image layer, so dotnet install will work.